libdnf5/utils/os_release.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "libdnf5/utils/os_release.hpp"
+
+#include <filesystem>
+#include <fstream>
+#include <regex>
+
+namespace libdnf5::utils {
+
+std::string OSRelease::get_value(const std::string & key, const std::string & default_value) {
+    initialize();
+    if (map.find(key) == map.end())
+        return default_value;
+    return map[key];
+}
+
+
+void OSRelease::initialize() {
+    if (initialized_)
+        return;
+
+    initialized_ = true;
+    std::ifstream infile(path);
+    if (!std::filesystem::exists(path))
+        return;
+
+    const std::regex r_no_quotes("^([A-Z_]+)=(\\w+)");
+    const std::regex r_quotes("^([A-Z_]+)=\"([\\w\\s]+)\"");
```

Comment:
I noticed some of our (currently disabled) CI tests would fail because of this regex.
For example if there is a value such as: `VERSION_ID="8.1"` in the `/ets/os-release` this regex doesn't match it.

Looking at the os-release.5 man page it seems that in general it can contain anything apart from non-printable characters. Some keys are further restricted to: no spaces or other characters outside of 0–9, a–z, ".", "_" and "-".
In any case this regex is way too strict.